designs/Transitive-Dependencies.md

```diff
@@ -0,0 +1,150 @@
+
+The Visual Studio Package Manager UI does not support the concept of showing transitive packages in any fashion.
+
+![](../resources/TransitiveDependencies/VSPMUI.png)
+
+## Success: How do we know if we have solved this problem?
+
+By measuring the number of detractors regarding transitive dependencies & promoters of this new experience in future NuGet client surveys.
+
+We will add telemetry to categorize when a top-level package is installed, and whether it was done in the context of promoting a transitive package to a top-level one.
+
+## Audience: Who are we building this for?
+
+We are building this for .NET users, both old and new, to be successful with understanding the types of NuGet dependencies in their project(s) and solution(s).
+
+## What: What does this look like in the product?
+
+![](../resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+## Context and Scope
+
+-	https://github.com/NuGet/Home/issues/5887 
+-	https://developercommunity.visualstudio.com/content/idea/538119/add-dependency-tree-view-for-packagereferences.html 
+
+## Minimal Requirements
+
+-	Visual list of transitive dependencies.
+-	Ability to see a transitive dependency package details.
+-	Ability to determine where a transitive dependency originated.
+-	Ability to promote a transitive dependency to a top-level dependency.
+
+## Scenarios and User Experience
+
+### CLI
+
+We should consider a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of where the package came from. This could be a new column next to “Resolved” which says “Transitively Referenced” or “Referenced” and has a list of the top-level packages that requested the dependency.
+
+![](../resources/TransitiveDependencies/TransitiveDotNetCLI.png)
+
+### Visual Studio
+
+Within Visual Studio, there will be a new panel within the Visual Studio Package Manager UI named “Transitive Packages” in which all transitive packages will be displayed to the user. For the sake of not confusing the user, there will additionally be a header titled “Top-level Packages” where the current experience is today for top-level packages.
+
+**By default, a user’s transitive packages will be collapsed** as a project can have hundreds, if not thousands of transitive packages & would potentially impact performance displaying these many records on each load.
+
+When a user highlights a transitive package, they will see a pop-up that displays how the transitive dependency originated & what top-level package(s) are bringing it in.
+
+![](../resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+Finally, selecting a transitive dependency will provide the user the ability to “Install” the package as you normally would a top-level dependency. This will promote the transitive dependency to a top-level dependency, and the transitive dependency will now be placed in the top-level packages list instead.
+
+## Goals and Non-Goals
+
+### Goals
```

Comment:
I found it weird that the goals are below the what. :) 

Maybe there was a meeting benefit here, but I think I was OOF that day.

Reply:
Haha, I just wanted to make sure people were reminded of the goals, but more importantly just to blabber non-goals 👍